In [30]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = 'all'

# Python - Question 1

In [57]:
list1 = [[1,2,3,4], [5,6,7,8], [9,10,11,12]]
list1[0][3]

4

In [42]:
for i in list1:
    print(i)

[1, 2, 3, 4]
[5, 6, 7, 8]
[9, 10, 11, 12]


In [59]:
for i in list1:
    if j in i:
        print("yes")

NameError: name 'j' is not defined

In [54]:
for i in range (len(list1)):
    print(i)

0
1
2


here we have to find the elements in each list item. above list contains 3 item lists. again in each item list, there are 4 more items. then we have to see if the tagret value is present here. 


#  Python Question 2

In [ ]:
string1 = input("Enter the string")

for item in string1:
    

In [ ]:
def freq():
    string = ["python", "hadoop",  "hadoop", "python",  "python", "scala" ,"python" ]
    

In [60]:
string1 = ["python", "hadoop",  "hadoop", "python",  "python", "scala" ,"python" ]
len(string1[0])

6

In [69]:
for i in range (len(string1)):
    print(i)

0
1
2
3
4
5
6


In [66]:
for j in string1:
    print (j)

python
hadoop
hadoop
python
python
scala
python


In [73]:
for i in string1:
    print(i,j)

ValueError: too many values to unpack (expected 2)

In [85]:
len("python")

6

In [86]:
string = ["python", "hadoop",  "hadoop", "python",  "python", "scala" ,"python" ]
len(string[0])

6

####  Explanation  
we should count the number of occurences of each word and that which occurs maximum times. 
here we find the word "Python" occuring 4 times and this is the maximum time a word has occured. 
then we should find length of python word, which is 6

# Machine Learning - Question 2

### SVM regressor on  
#### Bengaluru dataset

In [74]:
import pandas as pd
import numpy as np

In [81]:
data = pd.read_csv("Bengaluru_House_Data.csv")
data.head()

area_type   availability                  location       size  \
0  Super built-up  Area         19-Dec  Electronic City Phase II      2 BHK   
1            Plot  Area  Ready To Move          Chikka Tirupathi  4 Bedroom   
2        Built-up  Area  Ready To Move               Uttarahalli      3 BHK   
3  Super built-up  Area  Ready To Move        Lingadheeranahalli      3 BHK   
4  Super built-up  Area  Ready To Move                  Kothanur      2 BHK   

   society total_sqft  bath  balcony   price  
0  Coomee        1056   2.0      1.0   39.07  
1  Theanmp       2600   5.0      3.0  120.00  
2      NaN       1440   2.0      3.0   62.00  
3  Soiewre       1521   3.0      1.0   95.00  
4      NaN       1200   2.0      1.0   51.00

In [82]:
data.tail(5)

area_type   availability               location       size  \
13315        Built-up  Area  Ready To Move             Whitefield  5 Bedroom   
13316  Super built-up  Area  Ready To Move          Richards Town      4 BHK   
13317        Built-up  Area  Ready To Move  Raja Rajeshwari Nagar      2 BHK   
13318  Super built-up  Area         18-Jun        Padmanabhanagar      4 BHK   
13319  Super built-up  Area  Ready To Move           Doddathoguru      1 BHK   

       society total_sqft  bath  balcony  price  
13315  ArsiaEx       3453   4.0      0.0  231.0  
13316      NaN       3600   5.0      NaN  400.0  
13317  Mahla T       1141   2.0      1.0   60.0  
13318  SollyCl       4689   4.0      1.0  488.0  
13319      NaN        550   1.0      1.0   17.0

In [83]:
len(data)

13320

In [84]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13320 entries, 0 to 13319
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   area_type     13320 non-null  object 
 1   availability  13320 non-null  object 
 2   location      13319 non-null  object 
 3   size          13304 non-null  object 
 4   society       7818 non-null   object 
 5   total_sqft    13320 non-null  object 
 6   bath          13247 non-null  float64
 7   balcony       12711 non-null  float64
 8   price         13320 non-null  float64
dtypes: float64(3), object(6)
memory usage: 936.7+ KB


The dataset has 13,320 records.    
It has 7 independant variables  
It has 1 target variable which is PRICE  
society variable has many missing values  